Miner keeps panic when committing a specific sector

**Describe the bug**
Something wrong happens when committing a sector, which causes a panic (shutdown) of the miner. Then it restarts (since we use systemctl start lotus-miner). And the miner would do something (like windowPoSt) and then it tries to commit that sector again. But still, it will panic once again as the previous time. And this goes on and on forever.

And a side effect of this problem is that every-time the miner restarts, it will do a new windowedPoSt, over and over again. My miner ID is t01105, you might be able to see that it submit way too many windowedPoSt on chain (once or twice on each tipset height).

**To Reproduce**
Steps to reproduce the behavior:
No sure, happened to me three times already, also seen other people having the same problem

**Expected behavior**
1. We should have a way to save the miner from repeating panic (either declare the sector as faulty or remove the sector would be good, but currently neither of them works).  

2. The miner should not do repeating windowedPoSt whenever it restarts (though this issue is minor)

**Screenshots**
2020-08-14T22:06:48.218+0800    DEBUG   stores  stores/index.go:310     not selecting on 1d328abd-826c-4777-a6ea-746ca5480a51, out of space (available: 0, need: 8106750771)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x116ea36]
goroutine 56779 [running]:
github.com/filecoin-project/storage-fsm.(*Sealing).handleCommitting(0xc0120deb00, 0x2da1bc0, 0xc000040090, 0xc02d9d0c10, 0xc0297de270, 0xa, 0x86c, 0x2, 0xc0205c8860, 0x1, ...)
        /root/lotus/extern/storage-fsm/states_sealing.go:324 +0xda6
github.com/filecoin-project/storage-fsm.(*Sealing).Plan.func1(0x2da1bc0, 0xc000040090, 0xc02d9d0c10, 0xc0297de270, 0xa, 0x86c, 0x2, 0xc0205c8860, 0x1, 0x1, ...)
        /root/lotus/extern/storage-fsm/fsm.go:24 +0x81
reflect.Value.call(0x269cc40, 0xc02039a500, 0x13, 0x298de5b, 0x4, 0xc0123c9f70, 0x2, 0x2, 0x540dcf, 0x27925e0, ...)
        /usr/local/go/src/reflect/value.go:460 +0x8ab
reflect.Value.Call(0x269cc40, 0xc02039a500, 0x13, 0xc02c9d7f70, 0x2, 0x2, 0xc02af7de22, 0xc, 0x1e)
        /usr/local/go/src/reflect/value.go:321 +0xb4
github.com/filecoin-project/go-statemachine.(*StateMachine).run.func3(0xc02039a450, 0x2da1bc0, 0xc000040090, 0xc02d9d0c10, 0xc02039a460, 0xc0205ca960)
        /root/go/pkg/mod/github.com/filecoin-project/go-statemachine@v0.0.0-20200730031800-c3336614d2a7/machine.go:102 +0x2bf
created by github.com/filecoin-project/go-statemachine.(*StateMachine).run
        /root/go/pkg/mod/github.com/filecoin-project/go-statemachine@v0.0.0-20200730031800-c3336614d2a7/machine.go:100 +0x3da

**Version (run `lotus version`):**
4c5e9614 of ntwk-calibration

**Additional context**
Add any other context about the problem here.

--- a/extern/storage-sealing/states_sealing.go
+++ b/extern/storage-sealing/states_sealing.go
@@ -321,17 +321,17 @@ func (m *Sealing) handleCommitting(ctx statemachine.Context, sector SectorInfo) 
 		return nil
 	}
 
-	collateral, err := m.api.StateMinerInitialPledgeCollateral(ctx.Context(), m.maddr, *sector.PreCommitInfo, tok)
-	if err != nil {
-		return xerrors.Errorf("getting initial pledge collateral: %w", err)
-	}
-
 	pci, err := m.api.StateSectorPreCommitInfo(ctx.Context(), m.maddr, sector.SectorNumber, tok)
 	if err != nil {
 		return xerrors.Errorf("getting precommit info: %w", err)
 	}
 	if pci == nil {
 		return ctx.Send(SectorCommitFailed{error: xerrors.Errorf("precommit info not found on chain")})
+	}
+
+	collateral, err := m.api.StateMinerInitialPledgeCollateral(ctx.Context(), m.maddr, pci.Info, tok)
+	if err != nil {
+		return xerrors.Errorf("getting initial pledge collateral: %w", err)
 	}
 
 	collateral = big.Sub(collateral, pci.PreCommitDeposit)

